### ISSUE TITLE
Notes are not getting saved on MacOS Sierra 10.12.4 (16E195)

### ISSUE THREAD

**Issue by shivamb** [mstarke/MacPass#573]
I am using mac pass since 3 years and now in the latest MacOS update the notes are not getting saved.
When I insert any notes to an entry sand save the DB the notes will be available until the app is open/unlocked. When the app is closed and I reopen the DB the notes are gone. 
I tested in on another Mac and the issue remains the same.

MacPass: Version 0.6.2 (12420)
MacOS: MacOS Sierra 10.12.4 (16E195)

**Comment by sepehr** [comment]
It just hit me in the face. Lost a few of my credentials which were saved in the notes field.

**Comment by mstarke** [comment]
I'm deeply sorry for anyone losing data. This is a well known and fixed error. I am having not that much time so 0.7 is getting along but slowly. I think I will create an 0.6.3 to supply highly needed bug fixes to anyone using the binary distribution but without all the enhancements fully implemented. This then should reduce data loss for a considerable amount of users.

**Comment by mstarke** [comment]
Regarding the usage of the notes field:

A workaround for data loss is to use custom fields on the entries. This is not ideal since these aren't visible in the first tab but they are available for copy and paste via the context menu and also allow for better autotype support.

**Comment by sepehr** [comment]
@mstarke No worries Michael, I managed to recover the data. A 0.6.3 release would be great though; I'm looking forward to it.

Thank you for the quick reply, the workaround and your awesome work on MacPass.

**Comment by mstarke** [comment]
Duplicate of #445 #455 #463 #485 #552

**Comment by sebw** [comment]
Why is this getting closed? All reported duplicates are closed too.

As @shivamb I'm running Version 0.6.2 (12420) but on High Sierra (10.13).

As long as there is not a 0.6.3 with a fix, I don't see why these bugs are closed.

I just discovered this bug while trying to use 2 step auth backup codes for an account, only to realize there had never been saved.

**Comment by DominusExult** [comment]
It is being closed because 
- it is a duplicate of an existing filed issue. No need to keep duplicates around
- the bug IS fixed. There is no real release of the fixed code but the issue tracker is there to record bugs, so as soon as a bug is fixed it gets closed.

What you can do is use the continous build or download xcode and build it yourself (it's really not complicated at all, just takes time).
Opensource project, especially 1-man projects are in general not quick in releasing new versions as there are often other things that are considered vital for a new release. And since it is only one person working on it, this takes time.

**Comment by sebw** [comment]
This is no criticism. I'm just looking for an explanation and confirmation that MacPass got a patch for this critical issue.

In personal projects (not on Github) I mention bug reports in my patches/merge requests and they get closed automatically when the patch is merged. This is convenient and helps tracking how issues get resolved.

If the maintainer is not going to do that, people will keep reporting the same stuff over and over. It happened a couple of times for this issue, some people don't take time to look around for existing reports.. I did only to find at least 8 similar reports closed all marked duplicates and no mention of the patch or merge request, or release to come. It will be a waste of time for a one man project to go through triage.

For the sake of completeness and to help some people looking for information regarding this issue, these are the duplicates:

#445 
#455 
#463 
#485 
#552 
#539 
#544 
#573

**Comment by DominusExult** [comment]
When you look at the first report #445 you see the fix commit.
So as THIS issue is clearly marked as duplicate of #445, the real issue tracking is happening over there and not here.

**Comment by sebw** [comment]
Yes, I saw that commit from over a year now. No stable release including that patch since then, and how many reports? You're missing my point.

**Comment by mstarke** [comment]
@sebw thanks for the remarks. GitHub allows to close issues in commmits as well and I've done it before but rest asured that duplicates will arise all the time. You point to no officials release with the fix and rightly so but I am getting close to 0.7 and then a lot of this should be resolved. And I'm sorry for my occasional lack of documentation to make the development process more transparent.

**Comment by sebw** [comment]
Hey @mstarke no worries, I started using MacPass since seeing it gets mentioned on Hacker News every now and then. I'm checking for updates regularly and saw that no updates were available. The thing is, I lost a couple of notes, and see many other people in the same case.

I didn't know MacPass was a one man project, I'm sure many people would be happy to contribute to documentation and working on more frequent bugfix releases, esp. since you get mentionned on HN regularly.

Wish I could help but I don't develop on Mac.. :)

Cheers

**Comment by DominusExult** [comment]
@sebw It all boils down to a proper release and I wrote what usually happens with releases in 1man projects. You conveniently ignored that but instead made a lengthy post about tracking patches/commits/merges which Mstarke does quite well.
From my experience you will always get countless duplicates as people are not able to search issues and closed issues for some reason before adding an issue.
My way of doing things is looking at the date of last release, date of last commit and when there is a significant gap it's clear that you need to look more closely at open and closed issues to see if your issue has already been reported. This is hard for people it seems.

**Comment by DominusExult** [comment]
BUT IMO @mstarke should have made the old releases unavailable because of the very likely data loss

**Comment by sebw** [comment]
Your way of doing this is not the way of doing things of 95% of people who downloaded MacPass. I'm not going to go explain that to my gf. We are talking about data loss, which is a serious issue, while the app is saying there is no update for over a year.

Even if the old buggy release is made unavailable, it is currently installed on thousands of computers and marked as up-to-date, and users have no clue of the issue and will lose data.

As I'm saying above, this project needs more frequent bugfix releases, and if @mstarke can't keep up, he should turn to the community. I don't know what is included in 0.7 that makes it delay so long, but I'm not going to explain how bugfixes and features should be implemented, I don't want you to feel talked down to.

Anyway, I didn't intend offense towards the maintainer(s). My apologies if you felt offended.

**Comment by mstarke** [comment]
@sebw you hit some valid spots. Knowing that MacPass has bugs and might cause data loss for people is a bad feeling. I tried to find an excuse by providing open source licenses that allow anyone to contribute and fix bugs as well as add features and always stating the software is not fully stable (I'm calling them alpha for a reason). There were some really cool additions as well as awesome support for localisation and bug fixes in the past. The MacPassHTTP plugin for instance is just a port of code the community has contributed. But in the grand scheme the main work is done by me. And do not get me wrong, I like it a lot but there never was that much support from the community on the coding side. Again, I'm not trying to lessen the support and contributions since I'm very happy about anything contributed (I tend to include typos all over the place for instance). This makes me a bit jealous of the guys over at @keepassxreboot since there seems to be a lot more support. This might stem from my lack of a good contribution process or not actually encouraging…

Long story short - I do get the point about showing "up to date" in the dialog but actually serving software that is way older than it should be.
